node/narwhal/src/helpers/storage.rs

```diff
@@ -817,22 +817,258 @@ pub mod tests {
 #[cfg(test)]
 pub mod prop_tests {
     use super::*;
-    use crate::helpers::committee::prop_tests::CommitteeInput;
+    use indexmap::indexset;
+    use proptest::{
+        collection,
+        prelude::{any, BoxedStrategy, Just, Strategy},
+        prop_oneof,
+    };
+    use snarkvm::{ledger::coinbase::PuzzleCommitment, prelude::Uniform};
+    use std::fmt::Debug;
+    use test_strategy::proptest;
 
-    use test_strategy::Arbitrary;
+    use crate::MAX_GC_ROUNDS;
 
     type CurrentNetwork = snarkvm::prelude::Testnet3;
 
-    #[derive(Arbitrary, Debug, Clone)]
-    pub struct StorageInput {
-        #[filter(CommitteeInput::is_valid)]
-        pub committee: CommitteeInput,
-        pub gc_rounds: u64,
+    impl Arbitrary for Storage<CurrentNetwork> {
+        type Parameters = CommitteeContext;
+        type Strategy = BoxedStrategy<Storage<CurrentNetwork>>;
+
+        fn arbitrary() -> Self::Strategy {
+            (any::<CommitteeContext>(), 0..MAX_GC_ROUNDS)
+                .prop_map(|(CommitteeContext(committee, _), gc_rounds)| {
+                    Storage::<CurrentNetwork>::new(committee, gc_rounds)
+                })
+                .boxed()
+        }
+
+        fn arbitrary_with(context: Self::Parameters) -> Self::Strategy {
+            (Just(context), 0..MAX_GC_ROUNDS)
+                .prop_map(|(CommitteeContext(committee, _), gc_rounds)| {
+                    Storage::<CurrentNetwork>::new(committee, gc_rounds)
+                })
+                .boxed()
+        }
+    }
+
+    use crate::helpers::{now, storage::tests::assert_storage};
+
+    use ::bytes::Bytes;
+    use proptest::{prelude::Arbitrary, test_runner::TestRng};
+    use rand::{CryptoRng, Error, Rng, RngCore};
+
+    use snarkvm::{
+        ledger::narwhal::{Batch, Data},
+        prelude::Signature,
+    };
+
+    // The `proptest::TestRng` doesn't implement `rand_core::CryptoRng` trait which is required in snarkVM, so we use a wrapper
+    #[derive(Debug)]
+    pub struct CryptoTestRng(TestRng);
```

Comment:
Two main reasons:
- the `proptest` crate implements its own `rand::RngCore`  which it uses internally (e.g. `prop_perturb` returns a strategy where rng is always identical during shrinking). If we wanted to use our `TestRng` instead, we would need to add explicit seed generation into all places where we generate anything that needs `TestRng` to keep shrinking proceed deterministically - so more boilerplate. Also, we might miss out on future improvements on shrinking that `proptest`'s `rand::RngCore` impl is able to use because they control the rng state internally.
- `TestRng::fixed` outputs the seed, which is a lot of spam e.g. during shrinking.